src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#ifndef SRC_MESH_MFEMDATA_HPP_
+#define SRC_MESH_MFEMDATA_HPP_
+
+#include <set>
+#include <vector>
+
+#include "mfem.hpp"
+
+#include "axom/core.hpp"
+#include "redecomp/redecomp.hpp"
+
+#include "tribol/types.hpp"
+#include "tribol/common/Parameters.hpp"
+#include "tribol/mesh/MethodCouplingData.hpp"
+
+namespace tribol
+{
+
+/**
+ * @brief Facilitates transfer to submesh low-order refined mesh
+ *
+ * This class simplifies transferring 1) a variable (e.g. displacement) from a
+ * higher-order grid function on a coarse mesh to a low-order grid function on a
```

Comment:
I would maybe not mention whether a mesh is coarse or fine. A HO mesh is coarser than its LOR counterpart, but maybe here we just say that a HO grid function on a HO ParSubMesh transfer to a low order refined mesh? Here I don't know what "fine mesh" is referring to and this class seems to deal specifically with sub meshes not a generic "coarse" mesh.

If we are going to mention dual spaces we have to be more specific, but on the other hand, not everyone understands what this means. We can say that we can transfer primal fields, such as displacement or velocity, from HO to LOR, and then we can transfer grid functions in their dual space (i.e. integrated forms), such as the integrated contact nodal force, from a low-order grid function on a LOR mesh to a HO grid function on the HO submesh. 

The last comment is confusing saying the coarse mesh is intended to live on the ParSubMesh level, but isn't required. Just think about the clarity of this confusing/complex workflow. If there are other use cases maybe we elaborate on those later.

Reply:
Comment has been clarified.